src/main/java/uk/gov/gchq/palisade/rule/WrappedRule.java

```diff
@@ -85,19 +83,40 @@ public WrappedRule(final UnaryOperator<T> function) {
      * @param predicate the simple {@link Predicate} rule to wrap.
      */
     public WrappedRule(final Predicate<T> predicate) {
-        requireNonNull(predicate, WRAPPED_RULE_WAS_INITIALISED_WITH_NULL + PREDICATE_STRING);
-        this.predicate = predicate;
+        this.setPredicate(predicate);
     }
 
     @JsonCreator
     public WrappedRule(@JsonProperty("rule") final Rule<T> rule,
                        @JsonProperty("function") final UnaryOperator<T> function,
                        @JsonProperty("predicate") final Predicate<T> predicate) {
+        this.setRule(rule);
+        this.setFunction(function);
+        this.setPredicate(predicate);
+        checkNullCount(rule, function, predicate);
+    }
+
+    @Generated
+    public Rule<T> getRule() {
+        return rule;
+    }
+
+    @Generated
+    public void setRule(final Rule<T> rule) {
+        requireNonNull(rule);
         this.rule = rule;
+    }
+
+    @Generated
+    public void setFunction(final UnaryOperator<T> function) {
+        requireNonNull(function);
         this.function = function;
-        this.predicate = predicate;
+    }
 
-        checkNullCount(rule, function, predicate);
+    @Generated
+    public void setPredicate(final Predicate<T> predicate) {
+        requireNonNull(predicate);
+        this.predicate = predicate;
     }
 
```

Comment:
missing getFunction and getPredicate methods

Reply:
They were in the wrong place.  Now they have been moved.  Note they are of the special ilk of being annotated with @JsonTypeInfo and I have added the @Generated annotation.